### PR TITLE
completion of week 4 assessment

### PR DESCRIPTION
**Pull request by omarbowen01**

### CODE REVIEW

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js]: This setup is a little trickier than our usual Jest test.  There may be a few syntax things on your expect statement and make sure to be checking the expected _output_.  It looks like you are currently checking the initial input in your expect.arrayContaining method.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js]: Logic looks great and excellent work on your pseudocode!  You may have one small syntax error that might make your test fail.  Just make sure your variable name casing for randomnum matches when you reference it.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:81]: I don't see a function written out.  Your pseudocode looks like you should be able to get there.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:120]: Pseudocode looks great and will definitely get you to where you want to be!  Please write out the function to go with it and run your test.  Remember, we want to run the test and see that we are getting the right kind of fail.  Then we write the function and check we are passing.  Please make sure to paste the test results for both fail and pass from your terminal in pseudocode.

File: interview-questions.md
Comment (elysemontano) [inline interview-questions.md:43]: Great job in this section!

File: ruby-challenges.rb
Comment (elysemontano) [inline ruby-challenges.rb:25]: Awesome!!  Ruby also has a handy built-in method that will give a boolean if a number is even and if a number is odd.

File: ruby-challenges.rb
Comment (elysemontano) [inline ruby-challenges.rb:44]: Excellent!  Please make sure to leave your variables uncommented out.  Any part of the code needed to run the file should be left commented in.

File: ruby-challenges.rb
Comment (elysemontano) [inline ruby-challenges.rb:71]: String should be in the casing convention of snake_case since this will be used as a param in the method.  PascalCasing is reserved for when we refer to a class.

File: ruby-challenges.rb
Comment (elysemontano) [inline ruby-challenges.rb:82]: Your logic looks great, there are a few small syntax errors on this problem. On line 73, it looks like you are not comparing the values in your statement, I would recommend equality operator here.  There also may be an extra double quote just after #{word} on line 76.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Requesting some changes: Please make sure to add your functions to problems 2 and 3 in code challenges.  Also, do not merge until I approve this pull request.  You should be able to push the changes to the same branch and I will see the updated changes.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:28]: Test is really close.  May want to check the syntax for arrayContaining and also make sure your expected output is accurate in your test.  Looks like you put the input there.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:54]: Logic looks great!  Make sure to have the same casing for your variables when accessing them in your function.  Looks like randomnum is different from line 48 to 49.  Also, please make sure to leave code uncommented in your PR.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:90]: I like your logic in using the Math.min and Math.max with the spread operator.  A function will only return one thing though, so you would need to combine the 2 numbers into an array and return that array.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:71]: Syntax and logic looks good in the test.  Just make sure that your data type in the array matches what you are expecting.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:109]: Looks like your test is asking for numbers as strings instead of numbers. Keep an eye on the data types you are working with both input and output.

File: code-challenges.test.js
Comment (elysemontano) [inline code-challenges.test.js:129]: Definitely heading in the right direction.  You started a filter, but what do you want to filter out?  You mentioned checking index in your pseudocode which seems like that would be a good approach.

**Review by elysemontano — APPROVED** [review]
Great work Omar!  Your pseudocode does a really good job describing your approach.  When writing out the functions in the code challenges, it is always good to check back with your pseudocode to make sure you have done all the steps.  Also, what does your terminal say when running the tests?  Always keep a close eye on the data type you are working with for both input and outputs.

File: code-challenges.test.js
Comment (omarbowen01) [reply code-challenges.test.js:81]: function is there

File: code-challenges.test.js
Comment (omarbowen01) [reply code-challenges.test.js:120]: done

**Pull request merged**
